frame/composable-traits/src/staking_rewards.rs

```diff
@@ -128,12 +133,12 @@ impl<AccountId, AssetId, Balance: AtLeast32BitUnsigned + Copy, Epoch: Ord, Rewar
 	}
 
 	pub fn state(&self, epoch: &Epoch, epoch_start: Timestamp) -> PositionState {
-		if self.lock_date.saturating_add(self.lock_duration) < epoch_start {
-			PositionState::Expired
-		} else if self.reward_epoch_start > *epoch {
-			PositionState::Pending
-		} else {
-			PositionState::LockedRewarding
+		match self.lock {
+			Some(StakingLock { duration, .. })
```

Comment:
in theory could have penalty without lock (i.e. 0 lock). 
for zero duration lock not sure if mint_date has sence, may be that pair should be optional? overall mint date is not resonable until lock, so may be it is lock date?